tests/e2e/onboarding.test.ts

```diff
@@ -131,6 +140,134 @@ test('onboarding with a short code', async ({ page, getOnboardingData }) => {
+
+	await expect(page).toHaveURL(/\/onboarding\/github/)
+	expect(
+		page.getByText(new RegExp(`welcome aboard ${ghUser.primaryEmail}`, 'i')),
+	).toBeVisible()
+
+	// fields are pre-populated for the user
+	const usernameInput = page.getByRole('textbox', { name: /username/i })
+	expect(usernameInput).toHaveValue(
+		// @ts-ignore
+		normalizeUsername(ghUser.profile.login),
+	)
+	expect(page.getByRole('textbox', { name: /^name/i })).toHaveValue(
+		ghUser.profile.name,
+	)
+
+	// button currently in 'idle' (neutral) state and so has got no companion
+	const createAccountButton = page.getByRole('button', {
+		name: /create an account/i,
+	})
+	expect(createAccountButton.getByRole('status')).not.toBeVisible()
+	expect(createAccountButton.getByText('cross')).not.toBeAttached()
+
+	// attempt 1
+	// Username is a string of alphanums and underscores from 3 to 20 chars
+	// long, that will be lowercased upon sumbission. See: app/utils/user-validation.ts:
+	await usernameInput.fill('U$er_name') // $ is invalid char
+	await createAccountButton.click()
+
+	await expect(createAccountButton.getByRole('status')).toBeVisible()
+	expect(createAccountButton.getByText('cross')).toBeAttached()
+	await expect(
+		page.getByText(
+			/username can only include letters, numbers, and underscores/i,
+		),
+	).toBeVisible()
+	// but we also never checked that privacy consent box
+	await expect(
+		page.getByText(
+			/you must agree to the terms of service and privacy policy/i,
+		),
+	).toBeVisible()
+	await expect(page).toHaveURL(/\/onboarding\/github/)
+
+	// attempt 2
+	await usernameInput.fill('')
+	await createAccountButton.click()
+	await expect(page.getByText(/username is required/i)).toBeVisible()
+	// the next assertion will fail...
+	// await expect(createAccountButton.getByText('cross')).toBeAttached()
+	await expect(page).toHaveURL(/\/onboarding\/github/)
+
+	// attempt 3
+	await usernameInput.fill(`U5er_name_0k_${faker.person.lastName()}`)
+	await createAccountButton.click()
+	await expect(
+		page.getByText(/must agree to the terms of service and privacy policy/i),
+	).toBeVisible()
+	// the next assertion will fail...
+	// await expect(createAccountButton.getByText('cross')).toBeAttached()
+	await expect(page).toHaveURL(/\/onboarding\/github/)
+
+	// attempt 4 (we forgot about the checkbox)
+	await page
+		.getByLabel(/do you agree to our terms of service and privacy policy/i)
+		.check()
+	await createAccountButton.click()
+	// await expect(createAccountButton.getByText('cross')).not.toBeAttached()
+	await expect(page).toHaveURL(/signup/i) // home page
+
+	// we are still on the 'signup' route since that
+	// was the referrer and no 'redirectTo' has been specified
+	await expect(page).toHaveURL('/signup')
+	await expect(page.getByText(/thanks for signing up/i)).toBeVisible()
+
+	// internally, a user is being created:
+	const newlyCreatedUser = await prisma.user.findUniqueOrThrow({
```

Comment:
Is this a part of the testing setup (necessary for the next steps to pass) or an expected action in Prisma if the login flow reaches a certain state? 

Trying to figure out what this should do in relation to the test.

Reply:
This is to show a user in being created in out db, and I see what you mean, we never verified this user was not there already. Added a check

Reply:
Sorry, I'm still a bit confused. Let me phrase my question better:

- Is this database call a result of the user's interaction? Or,
- Is this database call required in order for future user interactions to even work correctly? 

Based on this, we need to approach this code differently.

Reply:
there are two calls: the first one demonstrates the user is not in our system, the other one (which we are running after they signup)  - shows they are in db. And yes - later on (at the clean up stage) we re-use the ref to the db user